hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/Storage.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.recordstore;
+
+import com.hazelcast.map.impl.SizeEstimator;
+
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * Represents actual storage layer behind a {@link RecordStore}.
```

Comment:
This doesn't add much value to me. The RecordStore is already the abstraction for record storage. You initially called it InternalRecordStore, but now you renamed to to another obscure name. What is the purpose of having another layer of indirection?